webapi/src/main/java/org/sakaiproject/webapi/controllers/RubricsRestController.java

```diff
@@ -370,7 +371,9 @@ public ResponseEntity<byte[]> getPdf(@PathVariable String siteId, @PathVariable
         checkSakaiSession();
 
         ContentDisposition contentDisposition = rubricsService.getRubric(rubricId).map(rubric -> {
-            return ContentDisposition.builder("attachment").filename(rubric.title + ".pdf").build();
+            String filename = StringUtils.trimToEmpty(rubric.title).replace(".", "_");
+            filename = StringUtils.isNotBlank(filename) ? filename : "_";
```

Comment:
I think this should throw an exception of some kind, or at least log something. A blank rubric title is not good - if blank rubric titles are possible, we should prevent that ever happening in the first place.

Reply:
I think we are on the same page, but just to be clear. This is not in the scope of the issue, right? With this PR I am trying to avoiding creating problematic filenames.

Reply:
Correct; on same page as separate but semi-related issues. I created https://sakaiproject.atlassian.net/browse/SAK-47400 to specifically address the blank name validation. Mentioned the weighted grading validation check as an example. In those instances, the **Save** button gets disabled until the issue is corrected so ideally that same behavior could occur with the title field.

FYI rubrics do get a default name of 'New rubric'. IMO the title field should start as blank and force an instructor to enter a name themself. Seen way too many instances of where an instructor doesnt notice that field and thus saves almost all their rubrics as 'New rubric'. Becomes more complicated when instructor then connects a rubric to an assignment and that locks in that rubric title.

Reply:
Thanks, perfect! :)
The described behavior for creating a rubric sounds good to me :+1: